File: AspNetMvcUnleashed/0672329980_source/Chapter07/VB/MvcApplication1/MvcApplication1/Controllers/Movie3Controller.vb
Public Class Movie3Controller
    Inherits System.Web.Mvc.Controller

    ' GET: /Movie3/Create
    <ActionName("Create")> _
    Function Create_GET() As ActionResult
        Return View()
    End Function

    ' POST: /Movie3/Create
    <AcceptVerbs(HttpVerbs.Post)> _
    <ActionName("Create")> _
    Function Create_POST() As ActionResult
        Dim movieToCreate As New Movie()
        Me.UpdateModel(movieToCreate, New String() {"Title", "Director", "DateReleased"})
        ' Insert movie into database
        Return RedirectToAction("Index")
    End Function


End Class

File: AspNetMvcUnleashed/0672329980_source/Chapter07/CS/MvcApplication1/MvcApplication1/Controllers/Movie3Controller.cs
using System.Web.Mvc;
using MvcApplication1.Models;

namespace MvcApplication1.Controllers
{
    public class Movie3Controller : Controller
    {
        // GET: /Movie3/Create
        [ActionName("Create")]
        public ActionResult Create_GET()
        {
            return View();
        } 

        // POST: /Movie3/Create
        [AcceptVerbs(HttpVerbs.Post)]
        [ActionName("Create")]
        public ActionResult Create_POST()
        {
            var movieToCreate = new Movie();
            this.UpdateModel(movieToCreate, new string[] {"Title", "Director", "DateReleased"});
            // Add movie to database
            return RedirectToAction("Index");
        }

    }
}
